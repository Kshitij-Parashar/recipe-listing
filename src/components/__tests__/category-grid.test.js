import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import CategoryGrid from '../categories';
import '../../setupTests';

it('should render correctly <CategoryGrid>', () => {
  const mockStore = configureStore();
  const store = mockStore({
    categories: {
      categories: {
        idCategory: '1',
        strCategory: 'Beef',
        strCategoryDescription:
          'Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]',
        strCategoryThumb: 'https://www.themealdb.com/images/category/beef.png',
      },
      loading: false,
      error: false,
    },
  });
  const wrapper = mount(
    <Provider store={store}>
      <CategoryGrid />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
