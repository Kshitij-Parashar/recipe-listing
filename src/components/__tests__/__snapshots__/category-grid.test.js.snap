// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly <CategoryGrid> 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(CategoryGridContainer)>
    <CategoryGridContainer
      categories={
        Object {
          "idCategory": "1",
          "strCategory": "Beef",
          "strCategoryDescription": "Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]",
          "strCategoryThumb": "https://www.themealdb.com/images/category/beef.png",
        }
      }
      error={false}
      loadCategories={[Function]}
      loading={false}
    >
      <h1
        className="heading center"
      >
        Categories
      </h1>
      <section
        className="categories"
      >
        <CategoryGrid
          categories={Array []}
          error={false}
          loading={false}
        >
          <h2>
            No category found.
          </h2>
        </CategoryGrid>
      </section>
      <Pagination
        currentPage={1}
        paginate={[Function]}
        postsPerPage={5}
        totalPosts={0}
      >
        <input
          className="hide"
          id="currentPage"
          onChange={[Function]}
          value={1}
        />
        <div
          className="pagination"
        >
          <button
            className="previous"
            id="prevPage"
            onClick={[Function]}
          >
            « Prev
          </button>
          <button
            className="next"
            id="nextPage"
            onClick={[Function]}
          >
            Next »
          </button>
        </div>
      </Pagination>
    </CategoryGridContainer>
  </Connect(CategoryGridContainer)>
</Provider>
`;
